Exercise_2.java

```diff
@@ -8,11 +8,23 @@
        of pivot */
     void swap(int arr[],int i,int j){
         //Your code here   
+        int temp = arr[i];
+        arr[i] = arr[j];
+        arr[j] = temp;
     }
     
     int partition(int arr[], int low, int high) 
     { 
    	//Write code here for Partition and Swap 
+       int smallIndex = low;
+       int pivot = high;
+       for(int i = low; i <= high; i++) {
+           if(arr[i] < arr[pivot]) {
+               swap(arr,i,pivot);

```

Comment:
swap I and smallIndex and increment smallIndex. Please go through the logic once again.